### ISSUE TITLE
Ability to enable all variants

### ISSUE THREAD

**Issue by benface** [tailwindcss/tailwindcss#851]
This used to be possible in pre-1.0.0, with `modules: 'all'`. Unfortunately `variants: 'all'` doesn't seem to work. I guess it should enable all variants (including custom ones added by plugins?) for all enabled core plugins (not sure if there's a way to extend it to non-core plugins as well, since they control the variants that get generated).

**Comment by adamwathan** [comment]
The way I'm thinking about solving this is by allowing something like:

```
variants: ['responsive', 'first-child', 'group-hover', 'hover', 'active', 'focus']
```

So no `all` keyword, but instead an array that applies those variants in that order to every plugin. The challenge is making this "just work" for third-party plugins, now that we encourage plugin authors to source their plugin variants from the user's config.

I'm thinking maybe a `variants` function we pass to the plugin that should be checked instead of `config`, so that if variants is an array we can return that array no matter what the requested key is?

For example:

```js
    function({ addUtilities, variants }) {
      const newUtilities = {
        '.skew-10deg': {
          transform: 'skewY(-10deg)',
        },
        '.skew-15deg': {
          transform: 'skewY(-15deg)',
        },
      }

      addUtilities(newUtilities, variants('skew'))
    }
```

Thoughts?

**Comment by benface** [comment]
@adamwathan That makes sense to me. Perhaps, just for consistency, there should also be a `theme` function, even though you can already do `config('theme.[key]')`? That way you can use `theme` and `variants` in the same way, with the plugin’s “handle”.

**Comment by adamwathan** [comment]
Solved in #852.
